tools/biosnoop.py

```diff
@@ -106,14 +106,12 @@
  * test, and maintenance burden.
  */
 #ifdef REQ_WRITE
-if (req->cmd_flags & REQ_WRITE) {
+    data.rwflag = !!(req->cmd_flags & REQ_WRITE);
+#elif defined(REQ_OP_SHIFT)
+    data.rwflag = !!((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE);
 #else
-if ((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE) {
+    data.rwflag = !!(req->cmd_flags & REQ_OP_WRITE);
```

Comment:
doesn't look right. It probably should be (req->cmd_flags & REQ_OP_MASK) == REQ_OP_WRITE